src/generate/apiCollection.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ApiGroup } from "./apiGroup";
+
+/**
+ * A collection of multiple groups of APIs stored by key.
+ */
+export class ApiCollection {
+  [key: string]: ApiGroup;
+
+  /**
+   * Loads an entire collection of APIs from a simple description format.
+   *
+   * @param description - a simple object literal with group names as
+   * properties and a list of file paths under each group
+   */
+  static async read(description: {
+    [key: string]: string[];
+  }): Promise<ApiCollection> {
+    const apiCollection = new ApiCollection();
+    for (const group in description) {
+      apiCollection[group] = await ApiGroup.read(description[group]);
```

Comment:
Do we have to set groups directly on the instance? Setting a property might be safer (avoids naming collisions with methods). Something like `apiCollection.collection = new Map()` or `apiCollection.collection = Object.create(null)`

Reply:
Additionally, if this change is made, then this class could also get `name`/`setName`, which I think would be useful.